Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/SpherePointer.cs

```diff
@@ -257,14 +289,116 @@ public bool TryUpdateQueryBufferForLayerMask(LayerMask layerMask, Vector3 pointe
 
                 for (int i = 0; i < numColliders; i++)
                 {
-                    grabbable = queryBuffer[i].GetComponent<NearInteractionGrabbable>();
+                    Collider collider = queryBuffer[i];
+                    grabbable = collider.GetComponent<NearInteractionGrabbable>();
+                    if (grabbable != null)
+                    {
+                        if (ignoreCollidersNotInFOV)
+                        {
+                            if (!isInFOVCone(collider))
+                            {
+                                // Additional check: is grabbable in the camera frustrum
+                                // We do this so that if grabbable is not visible it is not accidentally grabbed
+                                // Also to not turn off the hand ray if hand is near a grabbable that's not actually visible
+                                grabbable = null;
+                            }
+                        }
+                    }
+
                     if (grabbable != null)
                     {
                         return true;
                     }
                 }
                 return false;
             }
+
+
+            /// <summary>
+            /// Returns true if a collider's bounds is within the camera FOV
+            /// </summary>
+            /// <param name="myCollider">The collider to test</param>
+            private bool isInFOVCone(Collider myCollider)
```

Comment:
Can we reuse/extend this method:
https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk_development/Assets/MixedRealityToolkit/Utilities/MathUtilities.cs#L496 
or its camera extension:
https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk_development/Assets/MixedRealityToolkit/Extensions/CameraExtensions.cs#L34
or does this do something different?
Could we add it as a math utility if it might be useful elsewhere as well?

Reply:
Yes! I was planning to do that in a follow up PR. If I make it a public utility, I would need to add unit tests. Wanted to keep this PR scoped, and then in follow up PR add unit tests. Created #6950 to track the work.

Reply:
Sounds good!